### ISSUE TITLE
Nightly SLT: part encode task failed JoinError::Cancelled(Id(44940))

### ISSUE THREAD

**Issue by philip-stoev** [MaterializeInc/materialize#13930]
### What version of Materialize are you using?

50670b

### How did you install Materialize?

Docker image

### What is the issue?

Some sort of tokio issue I assume 

https://buildkite.com/materialize/sql-logic-tests/builds/5312#01823f6d-4fd5-44cd-b460-55bbd51323da

### Relevant log output

```source
thread 'tokio-runtime-worker' panicked at 'part encode task failed: JoinError::Cancelled(Id(44940))', /var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-0fba999a134df34b0-1/materialize/tests/src/persist-client/src/batch.rs:433:18
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
   4: <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll
   5: tokio::runtime::task::core::CoreStage<T>::poll
   6: tokio::runtime::task::harness::poll_future
   7: tokio::runtime::task::harness::Harness<T,S>::poll
   8: std::thread::local::LocalKey<T>::with
   9: tokio::runtime::thread_pool::worker::Context::run_task
  10: tokio::runtime::thread_pool::worker::Context::run
  11: tokio::macros::scoped_tls::ScopedKey<T>::set
  12: tokio::runtime::thread_pool::worker::run
  13: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
  14: tokio::runtime::task::harness::Harness<T,S>::poll
  15: tokio::runtime::blocking::pool::Inner::run
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by philip-stoev** [comment]
The hope is that this was fixed by recent tokio-related refactoring. Closing for the time being util it happens again.

**Comment by mjibson** [comment]
This is happening quite consistently in PRs and main builds now.

**Comment by danhhz** [comment]
Hopefully closed again by #14240
